scarlet/component.py

```diff
@@ -292,31 +317,39 @@ def coord(self):
             else:
                 return (self._index,)
 
-    def get_model(self, seds=None, morphs=None):
+    @property
+    def parameters(self):
+        parameters = []
+        for c in self.components:
+            parameters += c.parameters
+        return parameters
+
+    def get_model(self, *parameters):
         """Get the model this component tree
 
         Parameters
         ----------
-        seds: list of arrays
-            Optional list of seds for each component in the tree.
-            If `seds` is `None` then `self.sed` is used for
-            each component.
-        morphs: list of arrays
-            Optional list of morphologies for each component in
-            the tree. If `morphs` is `None` then `self.morph`
-            is used for each component.
+        parameters: list of arrays
+            Optional list of parameters for each component in the tree,
+            for example an `sed` and `morph` for each component.
+            If `parameters` is `None` then each component uses its
+            stored value for each parameter.
 
         Returns
         -------
         model: array
             (Bands, Height, Width) data cube
         """
         model = np.zeros((self.B, self.Ny, self.Nx))
-        for k in range(self.K):
-            if seds is not None and morphs is not None:
-                _model = self.components[k].get_model(seds[k], morphs[k])
+        pidx = 0
+        if len(parameters) > 0:
+            for k in range(self.K):
+                p = len(self.components[k].parameters)
+                _model = self.components[k].get_model(*parameters[pidx:pidx+p])
```

Comment:
I suggest `model += self.components[k].get_model(*parameters[pidx:pidx+p])` here

Reply:
You can't use an inline update with autograd, because it doesn't know that you made a change. See https://github.com/HIPS/autograd/blob/master/docs/tutorial.md#dont-use